### PR TITLE
convert to puma, add circle, Dockerfile

### PR DESCRIPTION
**Pull request by jlambert121**
Convert unicorn -> puma
Add `circle.yml`
Add Dockerfile

### CODE REVIEW

File: config/puma.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+preload_app!
+
+rackup      DefaultRackup
+port        ENV['PORT']     || 3000
+environment ENV['RACK_ENV'] || 'development'
+
+on_worker_boot do
+  # Worker specific setup for Rails 4.1+
+  # See: https://devcenter.heroku.com/articles/deploying-rails-applications-with-the-puma-web-server#on-worker-boot
+  if defined?(ActiveRecord::Base)
```
Comment (davetron5000) [inline config/puma.rb]: This app doesn't use a database, so this might not be needed.

Comment (davetron5000) [inline config/puma.rb]: Hmm, though I can see it was in the unicorn config, so let's leave it for "changing less things"

Comment (joecannatti) [inline config/puma.rb]: LOL, right. going to remove it

Comment (joecannatti) [inline config/puma.rb]: Sorry, didn't see your second comment. Will leave it

**Pull request merged**